### PR TITLE
fix(styleguide): Updates props table with variance support [WEB-1451]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

[Variance Migration Doc](https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194)

Dependencies (in order of merge)
1.  https://github.com/Codecademy/client-modules/pull/1691
2.  https://github.com/Codecademy/client-modules/pull/1658

As no `gamut-system` props are in use this switches the storybook system props addon to only support `variance` props and also cleans up the docgen stuff.

Unfortunately much of benefit of typescript docgen has been broken with typescript 4.2 as type aliases have much more caching and do not resolve to their literals.  I'm hoping there is a way to resolve this in the future but for now I've removed the docgen summaries from the table as they are confusing and not useful.

* Moves the PropsTable and related code inside the system addon directory and removes duplication between the two areas.  This also adds a preemptive new alias for `@codecademy/storybook-addon-variance` as it eventually should be split to its own package.
* Removes `react-typescript-docgen` summaries as they are no longer useful as of Typescript 4.2 due to improvements in type caching.
* Updates internal components for the PropsTable to no longer use `gamut-system`. 
* Refactors the enhancers to detect literal scale objects in addition to thematic scales.
* Small improvements to the type system inside of these components.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1451
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ffe600f7378eb9fba9ccb--gamut-preview.netlify.app>https://608ffe600f7378eb9fba9ccb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ffe600f7378eb9fba9ccb>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
💯

File: packages/styleguide/.storybook/addons/system/propMeta.ts [hunk 1]
```diff
@@ -1,59 +1,17 @@
...
-    return [...carry, ...handler.propNames];
+export type PropGroups = keyof typeof PROP_GROUPS;
+
+export const PROP_MAP = Object.entries(groups).reduce(
+  (carry, [key, handler]: [string, typeof groups[keyof typeof groups]]) => {
+    return { ...carry, ...handler.config };
   },
-  []
+  {} as Record<string, AbstractPropTransformer>
 );
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/addons/system/propMeta.ts:15]: So it's intentional that everything gets listed now? Seems a little verbose... can these be collapsed by default?

Screenshot of all sorts of px/mx/etc. prop spew

**Review by JoshuaKGoldberg — APPROVED** [review]

File: packages/styleguide/.storybook/addons/system/propMeta.ts (hunk 1 above)
Comment (codecaaron) [reply packages/styleguide/.storybook/addons/system/propMeta.ts:15]: That's the old one (sorry mix up).

New one correctly strips these out https://608ffe600f7378eb9fba9ccb--gamut-preview.netlify.app/storybook/?path=/docs/typography-text--text#code-playground

Comment (JoshuaKGoldberg) [reply packages/styleguide/.storybook/addons/system/propMeta.ts:15]: Oh awesome!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@36.12.1-alpha.ab776c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609164aec6bf14321499bdc0--gamut-preview.netlify.app>https://609164aec6bf14321499bdc0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609164aec6bf14321499bdc0>Deploy Logs</a></p>

**Pull request merged**